net/dns/manager.go

```diff
@@ -346,6 +348,125 @@ func (m *Manager) Query(ctx context.Context, bs []byte, from netaddr.IPPort) ([]
 	return m.resolver.Query(ctx, bs, from)
 }
 
+const (
+	// RFC 7766 6.2 recommends connection reuse & request pipelining
+	// be undertaken, and the connection be closed by the server
+	// using an idle timeout on the order of seconds.
+	idleTimeoutTCP = 45 * time.Second
+	// The RFCs don't specify the max size of a TCP-based DNS query,
+	// but we want to keep this reasonable. Given payloads are typically
+	// much larger and all known client send a single query, I've arbitrarily
+	// chosen 2k.
+	maxReqSizeTCP = 2048
+)
+
+// dnsTCPSession services DNS requests sent over TCP.
+type dnsTCPSession struct {
+	m *Manager
+
+	conn net.Conn
+	from netaddr.IPPort
+
+	responses chan []byte // DNS replies pending writing
+	queries   chan []byte // DNS requests from reading
```

Comment:
what's the point of this channel from the read side to the write side?

can't the reader start the handleQuery goroutines?

Reply:
Good point - fixed